### DISCUSSION TITLE
Formatting problem with v7 and MUI DateTimePicker

### DISCUSSION THREAD

**Discussion by sanderdesnaijer** [react-hook-form/react-hook-form#5360] [Q&A] [answered]
**Describe the bug**
When migrating from 6 to 7 we found one issue with our current MUI Controller wrappers, and that's with the DateTimePicker component; it does not format the value on mount. Only after focus or submitting the form it will show the correct format. When removing ```inputRef={field.ref}``` it will format correctly. 

**To Reproduce**
- open the sandbox and observe the unformatted value in the input. 
- when focus on input, or pressing submit it will render the correct format

**Codesandbox link (Required)**
https://codesandbox.io/s/mui-datepicker-focus-format-issue-0q687?file=/src/App.js

**Expected behavior**
It should format correctly on render. Is there a workaround or fix?

**Comment by bluebill1049** [comment]
hmmm, this is a tricky one... because by providing the ref of the input, hook form will update the input directly and a the same time change the input mask format. I think it's a limitation/compatibility with the library. I will leave out the `ref`, if the focus is an issue, I probably would use a separate focusable ref object. such as below:

```tsx
const inputRef = useRef();

inputRef.current = {
 target: {
   focus: e.target.focus
 }
}
```

**Reply by sanderdesnaijer** [reply]
Thanks for the quick reply. Not sure if I understand you correctly, but removing the ref indeeds solves the format issue. But when making the field empty, and pressing submit it won't focus on the field. Is there some kind of workaround for this?

**Comment by leapful** [comment, marked as answer]
@sanderdesnaijer 

The root cause is from Material-Ui date picker handling which always format date value before trigger "ref" callback of date input.

The suggestion from @bluebill1049 is to not register the real DOM input element but only focus method.

Another workaround: Trigger "register" before the formatting date from Material-UI.

Working sample: https://codesandbox.io/s/mui-datepicker-focus-format-issue-forked-ejwoc?file=/src/App.js

**Reply by sanderdesnaijer** [reply]
Thanks @leapful that worked!

**Reply by bluebill1049** [reply]
thank you @leapful 🙏

**Reply by bluebill1049** [reply]
That's really smart. @leapful you are a true hook form expert. 💪
